### ISSUE TITLE
Feature Request : Privacy zone

### ISSUE THREAD

**Issue by therealrobster** [zdila/gpx-animator#395]
Hi all,

I would love to have a privacy zone in the main settings.  I am adding a heap of tracks and each time I leave from  my house.  
It's super clear where I live. 
So I'd love to be able to drag a box, or enter co-ords of say, a 3 block radius around my area, that the rendering does not occur within.

So the tracks will be animated coming out of the box, and everywhere but within the box.  That way I can determine where I want to keep private, to reduce the chance of my actual house being known when sharing the renders.

I assume this does not exist, hence the feature request.

I hope this helps.  Just found gpx-animator today and am super impressed.  Thanks for all your efforts.

**Comment by McPringle** [comment]
Thank you very much for using GPX Animator and for your issue! It is not a drag-and-drop feature, but you can do this already with GPX Animator. Just specify, how much of your GPX track should not be part of your animation. The settings are called "Trim Start of GPX File" and "Trim End of GPX File". Just specify the duration of your ride, e.g. 10 minutes. GPX Animator will skip the first/last minutes of your track and your home or work address is kept private. This should solve your problem and answer your question. Enjoy GPX Animator! :-)

**Comment by therealrobster** [comment]
Thank you for this.  I've spent some time on that feature, now you've mentioned it.  
However, the issue is I will eventually have hundreds of tracks.  I'll need to manually edit every individual track's trim start / trim end.  To further complicate, it's quite hard to get the timing just right to maintain privacy.  It will take enormous effort.

What I'm doing for now, is writing a python script to try and remove any points within a lat/long bounding box.  That way I can do a :

for file in allMyGPXFiles:
    for track in gpx.tracks:
        for segment in track.segments:
            for point in segment.points:
                if within bounding box then delete this point.
etc

At least I can automate this.     The problem of course will be many start / stop areas all surrounding that bounding box.  Which is not a bad thing in itself, as it will create a sort of 'wall' of start / stop points.  What would be great though, which is why I asked about this feature, is if you could literally block it out with a coloured square (for example) with text overlayed ("Privacy Zone" for example).  That way people could clearly understand why the points are all starting and stopping around the exclusion zone.

Anyway, thank you again for this excellent software, I do appreciate it.  I hope the above helps to understand the limitations of the trim start / end and why I needed a little more.

I think my python script will suffice in the end, though not as elegant.

All the very best

**Comment by McPringle** [comment]
You are right, you showed me your use case much better. I understand your point of view and you convinced me to reopen this issue. A very easy first step could be to add a default trim to the preferences page, which is very easy and straightforward to implement. A second step could be to enter the latitude/longitude of the bounding box manually in the preferences, and when loading the GPX file remove the appropriate points.

But there could be more privacy zones than the own home address:

- work address(es)
- address from girl- or boyfriend
- addresses from other family members

So there should be a list of privacy zones...

- [ ] add a default trim to the preferences
- [ ] add one bounding box preference for privacy
- [ ] add more bounding boxes with names for privacy

**Comment by therealrobster** [comment]
Wow OK, thank you.  

I agree, there could I suppose be many exclusion / privacy zones, so to speak.  I wonder how that can be achieved interface wise.  My tracks, many of them cross the privacy zone, so to do it on each track would be painful.  That's for ME anyway.  For others that might work.  

I think a default trim would be a good starting place, on the main preferences as you say.  For the others, maybe it could be a list?  Select Zone1, settings appear.  Select Zone2, settings change to reflect Zone2, etc? 

I'm just not sure.  I do however, appreciate you re-opening this.

**Comment by therealrobster** [comment]
A further addition will be required I've just realised.  Even if there IS a privacy box, the LAT / LONG co-ords are still updating (presumably) in the text info area.  These would need to turn off / stop rendering, when the phone / GPS unit enters the privacy zone.  Otherwise it's essentially useless.  :-)  

Thanks again!
